src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -419,12 +479,54 @@ template<
 >
 template<typename Archive>
 void GAN<Model, InitializationRuleType, Noise, PolicyType>::
-serialize(Archive& ar, const unsigned int /* version */)
+serialize(Archive& ar, const unsigned int version)
 {
   ar & BOOST_SERIALIZATION_NVP(parameter);
   ar & BOOST_SERIALIZATION_NVP(generator);
   ar & BOOST_SERIALIZATION_NVP(discriminator);
-  ar & BOOST_SERIALIZATION_NVP(noiseFunction);
+
+  // Earlier versions of the GAN code did not serialize whether or not the model
+  // was reset.
+  if (version > 0)
+  {
+    ar & BOOST_SERIALIZATION_NVP(reset);
+    ar & BOOST_SERIALIZATION_NVP(genWeights);
+    ar & BOOST_SERIALIZATION_NVP(discWeights);
+  }
+  if (Archive::is_loading::value)
+  {
+    // The behavior in earlier versions was to always assume the weights needed
+    // to be reset.
+    if (version == 0)
```

Comment:
Since the code wasn't part of a stable release, I don't think we have to provide backward compatibility here.